pages/documentation/developer/tutorials/sensitivity-analysis.md

```diff
@@ -284,27 +284,28 @@ try (OutputStream os = Files.newOutputStream(jsonSensiResultPath)) {
 
 We now reach the last part of this tutorial, where we'll run a series of sensitivity calculations on a network, given a list of contingencies and sensitivity factors. Here we use the systematic sensitivity feature of Hades2, creating one variant on which all the calculations are done successively, without re-loading the network each time, by modifying the Jacobian matrix directly in the solver.
 
-First, we need to implement a contingencies provider.
 Here the list of contingencies is composed of the lines that are not monitored
 in the sensitivity analysis.
 ```java
-ContingenciesProvider contingenciesProvider = n -> n.getLineStream()
-  .filter(l -> {
-      final boolean[] isContingency = {true};
-      monitoredLines.forEach(monitoredLine -> {
-          if (l.equals(monitoredLine)) {
-            isContingency[0] = false;
-            return;
-          }
-      });
-      return isContingency[0];
-  })
-  .map(l -> new Contingency(l.getId(), new BranchContingency(l.getId())))
-  .collect(Collectors.toList());
+List<Contingency> contingencies = network.getLineStream()
+        .filter(l -> {
+            final boolean[] isContingency = {true};
+            monitoredLines.forEach(monitoredLine -> {
+                if (l.equals(monitoredLine)) {
+                    isContingency[0] = false;
+                    return;
+                }
+            });
+            return isContingency[0];
+        })
+        .map(l -> Contingency.builder(l.getId())
+                             .addBranch(l.getId())
+                             .build())
```

Comment:
Why don't you use the Contingency.branch() instead?
Please reduce the indentation to keep 4 spaces at every level.